wallet-api/src/Wallet/API.hs

```diff
@@ -100,76 +104,97 @@ $(deriveEq1 ''EventTriggerF)
 $(deriveOrd1 ''EventTriggerF)
 $(deriveShow1 ''EventTriggerF)
 
-type instance Base EventTrigger = EventTriggerF
+newtype AnnTriggerF a b = AnnTriggerF { getAnnTriggerF :: Compose ((,) a) EventTriggerF b }
```

Comment:
Wouldn't something like this be simpler? In particular I'd expect `AnnTriggerF` to actually be the functor for adding annotations, rather than the functor composed with `EventTriggerF`.
```
type AnnTriggerF a = (,) a
newtype EventTrigger a = EventTrigger { getEventTrigger :: Fix ((Compose (AnnTriggerF a) EventTriggerF) }
type instance Base (EventTrigger a) = Compose (AnnTriggerF a) EventTriggerF
```

Reply:
Yes that's better, thanks.